src/librustc/middle/infer/error_reporting.rs

```diff
@@ -486,13 +494,23 @@ impl<'a, 'tcx> ErrorReporting<'tcx> for InferCtxt<'a, 'tcx> {
             }
         };
 
+        let is_simple_error = if let &TypeError::Sorts(ref values) = terr {
```

Comment:
Why only make simple errors for TypeError::Sorts? Do other kinds of error never give the repetitive pattern?